### DISCUSSION TITLE
stripped down version for local sanboot only

### DISCUSSION THREAD

**Discussion by uli42** [ipxe/ipxe#1323] [General]
I am currently experimenting with a minimal ipxe that should only boot from local disk via `sanboot -drive 0` (embedded script). For this I don't need all that fancy networking stuff in the binary, I guess. I can disable a lot of features in general.h but I am wondering if there's a way to generate a `ipxe.efi` that only has the sanboot command and nothing else (also not autoboot). This would probably speed up download and startup of the `ipxe.efi`.

**Comment by NiKiZe** [comment]
I don't see this as something that we can do much with in terms of "issue" so moving to discussions.

Build a minimal set of drivers, such as snp.efi or snponly.efi. and see if that is good enough.

**Reply by uli42** [reply]
Thx, I will try the snponly.efi

Nevertheless I just found https://ipxe.org/appnote/named_config but I don't get it. Maybe someone can add some more explanation about the steps required to make an own configuration.

**Reply by NiKiZe** [reply]
Don't modify the existing .h files, instead modify the empty ones in the local directory...
Now you want to save the config, move them to a new directory, this is the named config, see the already existing named config directories for examples, such as vbox and other cloud ones.

**Reply by uli42** [reply]
Ah, thanks, can some please add this to the named config documentation link?

**Reply by NiKiZe** [reply]
I agree that it would maybe be good to mention that the local or your new named config should be used/modified and not the ones in the base config directory.

But reading the named config page again I don't think there is much to be done to explain that you create a new directory <name> and use that?

**Comment by uli42** [comment]
The problem ist that it is not clear what includes are quired there. Just
adding an example would help, I think. People not familiar with the build
system of ipxe will not immediately understand how all that is working.

BTW: can you spot where the system defines what is included in snponly and
what not?
